examples/flax/summarization/run_summarization_flax.py

```diff
@@ -337,24 +338,22 @@ def replicate(self):
 
 def data_loader(rng: jax.random.PRNGKey, dataset: Dataset, batch_size: int, shuffle: bool = False):
     """
-    Returns batches of size `batch_size` from truncated `dataset`, sharded over all local devices.
+    Returns batches of size `batch_size` from `dataset`. The final batch may be incomplete, and range in size from 1 to `batch_size`.
     Shuffle batches if `shuffle` is `True`.
     """
-    steps_per_epoch = len(dataset) // batch_size
+    steps_per_epoch = math.ceil(len(dataset) / batch_size)
 
     if shuffle:
         batch_idx = jax.random.permutation(rng, len(dataset))
+        batch_idx = np.asarray(batch_idx)
     else:
-        batch_idx = jnp.arange(len(dataset))
+        batch_idx = np.arange(len(dataset))
 
-    batch_idx = batch_idx[: steps_per_epoch * batch_size]  # Skip incomplete batch.
-    batch_idx = batch_idx.reshape((steps_per_epoch, batch_size))
+    batch_idx = np.array_split(batch_idx, steps_per_epoch)
 
     for idx in batch_idx:
         batch = dataset[idx]
-        batch = {k: jnp.array(v) for k, v in batch.items()}
-
-        batch = shard(batch)
```

Comment:
does `pad_shard_unpad` also takes care of sharding the batch ? (seems like it, but just to confirm)

Reply:
Yes sir!